### PR TITLE
Add groupchat application in examples

### PR DESCRIPTION
**Pull request by lakshminarayanagurram**

### CODE REVIEW

File: examples/groupchat.c [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+		set = meshlink_set_canonical_address(mesh, meshlink_get_self(mesh), host, port);
+
+		if(!set) {
+			fprintf(stderr, "Could not set canonical address '%s:%s': %s\n", host, port, meshlink_strerror(meshlink_errno));
+			return;
+		}
+
+		fprintf(stderr, "Canonical address set as '%s:%s'\n", host, port);
+	} else if(!strcasecmp(buf, "group")) {
```
Comment (gsliepen) [inline examples/groupchat.c]: The help text says the command is `/submesh`. You should either call it `group` everywhere, or `submesh` everywhere.

File: examples/groupchat.c [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+		if(!arg) {
+			nodes = meshlink_get_all_nodes(mesh, nodes, &nnodes);
+
+			if(!nnodes) {
+				fprintf(stderr, "Could not get list of nodes: %s\n", meshlink_strerror(meshlink_errno));
+			} else {
+				fprintf(stderr, "%zu known nodes:", nnodes);
+
+				for(size_t i = 0; i < nnodes; i++) {
+					fprintf(stderr, " %s", nodes[i]->name);
```
Comment (gsliepen) [inline examples/groupchat.c]: It would be nice to also print which submesh each node belongs to.

File: examples/groupchat.c (hunk 1 above)
Comment (lakshminarayanagurram) [reply examples/groupchat.c]: I have missed it. But now I have changed it to /group

File: examples/groupchat.c (hunk 2 above)
Comment (lakshminarayanagurram) [reply examples/groupchat.c]: In the latest commit iam also printing the submesh to which that node belongs to. For this i had to add the meshlink_get_all_nodes_by_submesh and meshlink_get_node_submesh api's in the latest commit

**Pull request merged**